### PR TITLE
Add gym-ignition project

### PR DESCRIPTION
**Pull request by diegoferigo**
- Add [`robotology/gym-ignition`](https://github.com/robotology/gym-ignition) to the [`core` profile](https://github.com/robotology/robotology-superbuild/blob/master/doc/profiles.md)
- Add new `ROBOTOLOGY_USES_IGNITION` option to enable code that depends on Ignition libraries

Remarks:

- The Ignition logic matches 1:1 the Gazebo Classic logic
- I added `gym-ignition` to the `core` profile since on the long term we could have something similar to `gazebo-yarp-plugins` for Ignition Gazebo, and Ignition support will likely be part of `core`
- `gym-ignition` depends on `idyntree`, therefore if `ROBOTOLOGY_USES_IGNITION=ON` and `ROBOTOLOGY_ENABLE_DYNAMICS=OFF`, also `idyntree` and all its dependent projects will be installed
- CI tests `ROBOTOLOGY_USES_IGNITION` only in conda
- macOS and Windows are not tested in conda due to https://github.com/conda-forge/libignition-gazebo-feedstock/issues/2

TODOs:

- https://github.com/robotology/robotology-superbuild/blob/85d1ea32192cfb7f026715b7c5b750a77f44a06f/cmake/Buildgym-ignition.cmake#L21

### CODE REVIEW

**Comment by traversaro** [comment]
* [ ] Can you add the latest release to https://github.com/robotology/robotology-superbuild/blob/master/releases/latest.releases.yaml ? 
* [ ] Can you update the profiles.md table specifying where `ROBOTOLOGY_USES_IGNITION` is supported (i.e. tested by CI)

**Comment by diegoferigo** [comment]
>     * [x]  Can you add the latest release to https://github.com/robotology/robotology-superbuild/blob/master/releases/latest.releases.yaml ?
> 
>     * [x]  Can you update the profiles.md table specifying where `ROBOTOLOGY_USES_IGNITION` is supported (i.e. tested by CI)

Done in https://github.com/robotology/robotology-superbuild/compare/85d1ea32192cfb7f026715b7c5b750a77f44a06f..6a3e8b6b3ff38756781c575274145dc19f5aff8b

File: cmake/Buildgym-ignition.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ycm_ep_helper(gym-ignition TYPE GIT
+              STYLE GITHUB
+              REPOSITORY robotology/gym-ignition.git
+              TAG master
+              COMPONENT core
+              FOLDER src
+              CMAKE_ARGS -DSCENARIO_USE_IGNITION:BOOL=${ROBOTOLOGY_USES_IGNITION}
+                         -DSCENARIO_ENABLE_BINDINGS:BOOL=${ROBOTOLOGY_USES_PYTHON}
+                         -DBINDINGS_INSTALL_PREFIX:PATH=${ROBOTOLOGY_SUPERBUILD_PYTHON_INSTALL_DIR}
+              DEPENDS iDynTree ign-gazebo5)
```
Comment (diegoferigo) [inline cmake/Buildgym-ignition.cmake:19]: @traversaro is the `ign-gazeboX` dependency OK? (There's a typo and this should be 4 for the current `master`, but I'm not sure that adding it is necessary)

Comment (traversaro) [reply cmake/Buildgym-ignition.cmake:19]: I typically set `DEPENDS`  only to the value of other projects that are handled by the superbuild. I am not really sure how it works for projects not handled by the superbuild, so if it works I would just leave iDynTree there.

File: doc/profiles.md [hunk 2]
```diff
@@ -228,6 +230,17 @@ the correct enviroment variables as documented in [`robotology-superbuild-depend
...
+## Ignition
+Support for this dependency is enabled by the `ROBOTOLOGY_USES_IGNITION` CMake option.
+This option is set to `OFF` on all platforms as it is still experimental.
+
+### System Dependencies
+Follow the official instructions to install Ignition on your platform, available at https://ignitionrobotics.org/docs.
+
+Different Ignition distributions can be installed alongside. 
+The projects included in the superbuild might require different distributions. 
+From the superbuild point of view, we currently do not allow enabling projects that only support a specific Ignition distribution, therefore all required distributions have to be found in the system.
```
Comment (traversaro) [inline doc/profiles.md:242]: If we support only Conda/Linux for `ROBOTOLOGY_USES_IGNITION`, we should describe here how to install Ignition Gazebo on conda (preferably the version tested in CI), as the official docs of Ignition Gazebo do not cover that the installation in conda.

Comment (diegoferigo) [reply doc/profiles.md]: Done in https://github.com/robotology/robotology-superbuild/compare/6a3e8b6b3ff38756781c575274145dc19f5aff8b..b76fbfe3a2dc0529aa5d7e9dc32dcced200f03fa

File: cmake/Buildgym-ignition.cmake (hunk 1 above)
Comment (diegoferigo) [reply cmake/Buildgym-ignition.cmake]: Remove in https://github.com/robotology/robotology-superbuild/compare/6a3e8b6b3ff38756781c575274145dc19f5aff8b..b76fbfe3a2dc0529aa5d7e9dc32dcced200f03fa

**Comment by diegoferigo** [comment]
CI failures don't seem related to this PR.

<details>
<summary>CI failure message</summary>

```
FAILED: src/yarp-matlab-bindings/CMakeFiles/YCMStamp/yarp-matlab-bindings-configure 
cd /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/yarp-matlab-bindings && /usr/share/miniconda/envs/test/bin/cmake --no-warn-unused-cli "-DCMAKE_PREFIX_PATH:PATH=/home/runner/work/robotology-superbuild/robotology-superbuild/build/install;/usr/share/miniconda/envs/test;/usr/share/miniconda/envs/test/x86_64-conda-linux-gnu/sysroot/usr" -DMatlab_ROOT_DIR:PATH=/home/runner/work/robotology-superbuild/robotology-superbuild/msdk_R2020b_mexa64 -DMatlab_MEX_EXTENSION:STRING=mexa64 -DYARP_USES_MATLAB:BOOL=ON -DYARP_USES_OCTAVE:BOOL=OFF -C/home/runner/work/robotology-superbuild/robotology-superbuild/build/src/yarp-matlab-bindings/CMakeFiles/YCMTmp/yarp-matlab-bindings-cache-Release.cmake -GNinja /home/runner/work/robotology-superbuild/robotology-superbuild/src/yarp-matlab-bindings && /usr/share/miniconda/envs/test/bin/cmake -E touch /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/yarp-matlab-bindings/CMakeFiles/YCMStamp/yarp-matlab-bindings-configure
Not searching for unused variables given on the command line.
loading initial cache file /home/runner/work/robotology-superbuild/robotology-superbuild/build/src/yarp-matlab-bindings/CMakeFiles/YCMTmp/yarp-matlab-bindings-cache-Release.cmake
-- Found YARP: /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP (found version "3.4.3+38-20210330.6+gitf7f000f39")
CMake Deprecation Warning at /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP/YARPConfig.cmake:226 (message):
  The YARP_INCLUDE_DIRS variable is deprecated
Call Stack (most recent call first):
  CMakeLists.txt:9223372036854775807 (_yarp_deprecated_variable_warning)
  CMakeLists.txt:29 (get_target_property)


-- Found YCM: /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/share/cmake/YCM (found version "0.12.1.12-20210301.4+gitdc98072")
CMake Deprecation Warning at /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/lib/cmake/YARP/YARPConfig.cmake:226 (message):
  The YARP_INCLUDE_DIRS variable is deprecated
Call Stack (most recent call first):
  CMakeLists.txt:9223372036854775807 (_yarp_deprecated_variable_warning)
  CMakeLists.txt:42 (include_directories)


CMake Error at cmake/FindMatlab.cmake:1197 (list):
  list index: 1 out of range (-1, 0)
Call Stack (most recent call first):
  matlab/CMakeLists.txt:96 (find_package)


-- Found Matlab: /home/runner/work/robotology-superbuild/robotology-superbuild/msdk_R2020b_mexa64/extern/include (found version "/home/runner/work/robotology-superbuild/robotology-superbuild/msdk_R2020b_mexa64") found components: MX_LIBRARY 
CMake Deprecation Warning at /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:314 (message):
  TARGETS is deprecated.  Use EXPORT instead
Call Stack (most recent call first):
  matlab/CMakeLists.txt:168 (install_basic_package_files)


CMake Deprecation Warning at /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/share/YCM/modules/InstallBasicPackageFiles.cmake:385 (message):
  Compatibility variables are no longer generated.  Use
  ENABLE_COMPATIBILITY_VARS to re-enable them (deprecated) or define them
  using either INCLUDE_FILE or INCLUDE_CONTENT (recommended).
Call Stack (most recent call first):
  matlab/CMakeLists.txt:168 (install_basic_package_files)


-- Configuring incomplete, errors occurred!
See also "/home/runner/work/robotology-superbuild/robotology-superbuild/build/src/yarp-matlab-bindings/CMakeFiles/CMakeOutput.log".
```

</details>

**Comment by traversaro** [comment]
> 
> 
> CI failures don't seem related to this PR.
> CI failure message

Are you sure? All other CI builds are working fine. Are you sure that it is not due to the `cmake --build build/ -DROBOTOLOGY_USES_IGNITION:BOOL=ON` command? Not that this does not configure again the cmake project, but rather builds it.

**Comment by diegoferigo** [comment]
> > CI failures don't seem related to this PR.
> > CI failure message
> 
> Are you sure? All other CI builds are working fine. Are you sure that it is not due to the `cmake --build build/ -DROBOTOLOGY_USES_IGNITION:BOOL=ON` command? Not that this does not configure again the cmake project, but rather builds it.

Checking the logs you're right, not sure why my local setup just configures it. I tried it locally before committing :man_shrugging:

**Comment by traversaro** [comment]
It seems good to go now, thanks (even if I would like to solve https://github.com/conda-forge/libignition-gazebo-feedstock/issues/4 before advertising it).

**Review by traversaro — APPROVED** [review]

**Review by traversaro — APPROVED** [review]

**Pull request merged**